src/ck-core/cklocation.C

```diff
@@ -1743,22 +1748,21 @@ void CkMigratable::ReadyMigrate(bool ready)
 }
 
 
-void CkMigratable::staticResumeFromSync(void* data)
+void CkMigratable::ResumeFromSyncHelper()
 {
-	CkMigratable *el=(CkMigratable *)data;
-	DEBL((AA "Element %s resuming from sync\n" AB,idx2str(el->thisIndexMax)));
+  DEBL((AA "Element %s resuming from sync\n" AB,idx2str(thisIndexMax)));
 
   if (_lb_args.metaLbOn()) {
-  	el->clearMetaLBData();
-	}
+    clearMetaLBData();
+  }
 
-  CkLocMgr *localLocMgr = el->myRec->getLocMgr();
-  auto iter = localLocMgr->bufferedActiveRgetMsgs.find(el->ckGetID());
+  CkLocMgr *localLocMgr = myRec->getLocMgr();
+  auto iter = localLocMgr->bufferedActiveRgetMsgs.find(ckGetID());
   if(iter != localLocMgr->bufferedActiveRgetMsgs.end()) {
-    std::pair<CmiUInt8, CkMigratable*> idElemPair(el->ckGetID(), el);
+    std::pair<CmiUInt8, CkMigratable*> idElemPair(ckGetID(), this);
```

Comment:
These two lines should be replaced by the following:

localLocMgr->toBeResumeFromSynced.emplace(std::make_pair(ckGetID(), this));